src/config.rs

```diff
@@ -5,13 +9,65 @@ pub trait Config {
 
     fn current_shell_name(&self) -> String;
 
-    fn set_current_shell_name(&mut self, name: &str);
+    fn set_current_shell_name(&mut self, name: &str) -> io::Result<()>;
 
-    fn does_shell_exist(&self, name: &str) -> bool;
+    fn does_shell_exist(&self, name: &str) -> io::Result<bool>;
+}
+
+#[derive(Clone)]
+pub struct FsConfig {
+    pub root_path: PathBuf,
+    pub current_shell: String,
+}
+
+impl FsConfig {
+    fn new(root_path: PathBuf) -> io::Result<Self> {
+        let config_path = root_path.join("current_shell");
+        let config_display = config_path.display();
+
+        let mut file = try!(File::open(&config_path));
+        let mut current_shell = String::new();
+
+        try!(file.read_to_string(&mut current_shell));
+
+        Ok(FsConfig {
+            root_path: root_path,
+            current_shell: current_shell,
+        })
+    }
+}
+
+impl Config for FsConfig {
+    fn root_path(&self) -> &PathBuf {
+        &self.root_path
+    }
+
+    fn current_shell_name(&self) -> String {
+        self.current_shell.clone()
```

Comment:
So I know I set the type for this return value.  But I'm thinking that maybe it makes more sense for it to be a `&String`? That way it can be borrowed by the hermit and the shell structs...  Not sure how that's going to work out in practice so let's skip it for now.